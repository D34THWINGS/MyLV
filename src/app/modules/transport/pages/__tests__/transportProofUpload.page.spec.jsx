import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import { TransportProofPage } from '../transportProofUpload.page'

jest.unmock('../transportProofUpload.page')

describe('TransportProofPage', () => {
  let props

  const getWrapper = () => shallow(<TransportProofPage {...props} />)

  beforeEach(() => {
    props = {
      classes: {},
      valid: false,
      handleSubmit: jest.fn(),
      setExpirationDateToCurrentMonth: jest.fn(),
      hasInvalidTransportProof: false,
    }
  })

  it('should render properly', () => {
    // When
    const wrapper = getWrapper()

    // Then
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
