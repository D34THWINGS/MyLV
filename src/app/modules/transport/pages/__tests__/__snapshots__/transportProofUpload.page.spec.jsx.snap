// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransportProofPage should render properly 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <Card
    raised={false}
  >
    <WithStyles(CardContent)>
      <WithStyles(GridWrapper)
        container={true}
      >
        <WithStyles(GridWrapper)
          item={true}
          xs={12}
        >
          <WithStyles(Typography)
            component="h2"
            gutterBottom={true}
            variant="headline"
          >
            Uploader un jsutificatif de titre de transport
          </WithStyles(Typography)>
          <WithStyles(Typography)
            gutterBottom={true}
          >
            Ce justificatif te permettra d'être remboursé pour ton titre de transport, merci donc de le fournir avant le 25 du mois. Renseigne aussi la date d'expiration du justificatif afin d'être rappellé automatiquement lorsque celui-ci arrive à expiration.
          </WithStyles(Typography)>
        </WithStyles(GridWrapper)>
        <WithStyles(GridWrapper)
          item={true}
          md={4}
          xs={12}
        >
          <Field
            autoFocus={true}
            component={[MockFunction]}
            fullWidth={true}
            label="Date de début de validité"
            name="startingDate"
            type="text"
          />
        </WithStyles(GridWrapper)>
        <WithStyles(GridWrapper)
          item={true}
          md={4}
          xs={12}
        >
          <Field
            component={[MockFunction]}
            fullWidth={true}
            label="Date d'expiration"
            name="expirationDate"
            type="text"
          />
        </WithStyles(GridWrapper)>
        <WithStyles(GridWrapper)
          item={true}
          md={4}
          xs={12}
        >
          <div>
            <WithStyles(Button)
              color="primary"
              onClick={[MockFunction]}
            >
              Valide pour ce mois
            </WithStyles(Button)>
          </div>
        </WithStyles(GridWrapper)>
        <WithStyles(GridWrapper)
          item={true}
          xs={12}
        >
          <Field
            accept="jpg,jpeg,JPG,JPEG,pdf,PDF,png,PNG"
            component={[MockFunction]}
            label="Choisir un fichier"
            name="file"
          />
        </WithStyles(GridWrapper)>
      </WithStyles(GridWrapper)>
    </WithStyles(CardContent)>
    <WithStyles(CardActions)>
      <WithStyles(Button)
        color="primary"
        disabled={true}
        size="small"
        type="submit"
      >
        Upload
      </WithStyles(Button)>
    </WithStyles(CardActions)>
  </Card>
</form>
`;
